### DISCUSSION TITLE
Can't use customFields based on process.env in markdown (.md/.mdx) files?

### DISCUSSION THREAD

**Discussion by boomerbgit** [facebook/docusaurus#7814] [Q&A] [answered]
I'm using basic starter site 'npx create-docusaurus...'
Updated docusaurus.config.js to have
```
  title: process.env.TITLE, 
  tagline: process.env.TAGLINE, 
  url: process.env.URL, 
...
  customFields: {
    myURL: process.env.URL
  },
```
With those system variables defined in the session where I'm running 'npm start' I can reference siteConfig.customFields.myURL in src/pages/index.tsx
but not in docs/intro.mdx (I renamed intro.md to intro.mdx)
In fact, just importing the docusaurus file that contains 'process.env.URL' fails.

Adding this line `import config from '@site/docusaurus.config.js';` to the docs/intro.mdx file causes the page to fail to load and logs a console error 
```
docusaurus.config.js?bb44:16 Uncaught (in promise) ReferenceError: process is not defined
    at eval (docusaurus.config.js?bb44:16:1)
    at ./docusaurus.config.js (content---docs-introf-84-2b4.js:22:1)
    at __webpack_require__ (runtime~main.js:36:33)
    at fn (runtime~main.js:332:21)
    at eval (intro.mdx:13:84)
```

I have verified that I can use (docusaurus.config.js)
```
 customFields: {
    myURL: 'myURL as string'
  },
```
and (docs/intro.mdx )
```
import config from '@site/docusaurus.config.js';
...
<div>Title: {config.title}</div>
<div>myURL: {config.customFields.myURL}</div>
```

**Comment by Josh-Cena** [comment, marked as answer]
Don't import from `@site/docusaurus.config.js`—it supposed to be run on server side. You can read https://docusaurus.io/docs/advanced/architecture. Use `@generated/docusaurus.config` instead.

**Reply by boomerbgit** [reply]
glad I asked that question.   I followed that link from your Deployment page on environment variables but from that picture and the description below I didn't pick up to change
```
@site/docusaurus.config.js
```
to
```
@generated/docusaurus.config
```

Thank you
